tests/test_ec2/test_elastic_network_interfaces.py

```diff
@@ -1040,12 +936,46 @@ def test_elastic_network_interfaces_describe_attachment():
         == "An error occurred (DryRunOperation) when calling the DescribeNetworkInterfaceAttribute operation: Request would have succeeded, but DryRun flag is set"
     )
 
-    my_eni_description = client.describe_network_interface_attribute(
+    my_eni_description = ec2client.describe_network_interface_attribute(
         NetworkInterfaceId=eni_id, Attribute="description"
     ).get("Description")
     assert my_eni_description["Value"] == "A network interface"
 
-    my_eni_source_dest_check = client.describe_network_interface_attribute(
+    my_eni_source_dest_check = ec2client.describe_network_interface_attribute(
         NetworkInterfaceId=eni_id, Attribute="sourceDestCheck"
     ).get("SourceDestCheck")
     assert my_eni_source_dest_check["Value"] is True
+
+
+@mock_aws
+@pytest.mark.skipif(settings.TEST_SERVER_MODE, reason="Fails in servermode")
+def test_eni_detachment():
+    # Setup
+    ec2resource, ec2client, vpc, subnet = setup_vpc(boto3)
+    ec2client.run_instances(ImageId="ami-12c6146b", MinCount=1, MaxCount=1)
```

Comment:
```suggestion
    resp = ec2client.run_instances(ImageId="ami-12c6146b", MinCount=1, MaxCount=1)["Instances"][0]
    eni_id = resp["NetworkInterfaces"][0]["Attachment"]["AttachmentId"]

    # Execute
    with pytest.raises(ClientError) as ex:
        ec2client.detach_network_interface(AttachmentId=eni_id)
```

Because the tests are run in parallel, and other ENI's exist, we can't just call `describe_network_interfaces` and get the first one - because that is not guaranteed to be ours.
We can get our ENI from the `run_instances`-call though.

(Note that I'm adding this as a suggestion, but I haven't formatted/linted this... )

Reply:
🤘